### PR TITLE
[fix] : always cleanup child clusters

### PR DESCRIPTION
**Pull request by rahulait**
**What this PR does / why we need it**:
This PR adds target in makefile to clean all child clusters. This will allow us to clean up resources when an e2e test run gets cancelled. We plan to always run the cleanup job to clean all child clusters when e2e test run finishes or gets cancelled.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -264,6 +264,15 @@ clean-release-git: ## Restores the git files usually modified during a release
 clean-release: clean-release-git
 	rm -rf $(RELEASE_DIR)
 
+.PHONY: clean-child-clusters
+clean-child-clusters: $(KUBECTL)
+	@for cluster in $$($(KUBECTL) get clusters -o name); do \
```
Comment (AshleyDumaine) [inline Makefile:269]: Should be able to do this I think?
`$(KUBECTL) delete clusters -A --all`
Same below for VPCs

**Comment by rahulait** [comment]
Did a test run and tried cancelling already running GHA, I can now see cleanup happening even if existing GHA run is cancelled.
```
Run make clean-child-clusters
  make clean-child-clusters
  shell: /usr/bin/bash -e {0}
  env:
    GITHUB_TOKEN: ***
    LINODE_TOKEN: ***
curl -fsSL https://dl.k8s.io/release/v1.[2](https://github.com/linode/cluster-api-provider-linode/actions/runs/10303706154/job/28521398443#step:8:2)8.0/bin/linux/amd64/kubectl -o /home/runner/work/cluster-api-provider-linode/cluster-api-provider-linode/bin/kubectl
chmod +x /home/runner/work/cluster-api-provider-linode/cluster-api-provider-linode/bin/kubectl
/home/runner/work/cluster-api-provider-linode/cluster-api-provider-linode/bin/kubectl delete clusters -A --all --timeout=180s
cluster.cluster.x-k8s.io "e2e-lm-vpc-1[6](https://github.com/linode/cluster-api-provider-linode/actions/runs/10303706154/job/28521398443#step:8:6)d2d21" deleted
cluster.cluster.x-k8s.io "e2e-min-lm-16d2d21" deleted
/home/runner/work/cluster-api-provider-linode/cluster-api-provider-linode/bin/kubectl delete linodevpc -A --all --timeout=60s
linodevpc.infrastructure.cluster.x-k[8](https://github.com/linode/cluster-api-provider-linode/actions/runs/10303706154/job/28521398443#step:8:9)s.io "e2e-lm-vpc-16d2d21" deleted
```

File: Makefile (hunk 1 above)
Comment (rahulait) [reply Makefile]: Thanks, updated.

**Review by AshleyDumaine — APPROVED** [review]
LGTM

**Pull request merged**